File: code/java/剑指offer22.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode node = head;
        int i = 1;
        while(node.next!=null&&i<k){
            node = node.next;
            i+=1;
        }
        if(i<k){
            return null;
        }
        ListNode target = head;
        while(node.next!=null){
            node = node.next;
            target = target.next;
        }
        return target;
    }
}

File: code/python/剑指offer22.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        if head is None:
            return None
        length = 0
        result = None
        temp = head
        while head is not None:
            length+=1
            if length>=k:
                if result is None:
                    result = temp
                else:
                    result = result.next
            #print(length,result)
            head = head.next
        return result